address_restricted.go

```diff
@@ -0,0 +1,126 @@
+package iotago
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/iotaledger/hive.go/lo"
+	"github.com/iotaledger/hive.go/serializer/v2"
+	"github.com/iotaledger/iota.go/v4/hexutil"
+)
+
+type RestrictedAddress struct {
+	Address             Address                    `serix:"0,mapKey=address"`
+	AllowedCapabilities AddressCapabilitiesBitMask `serix:"1,mapKey=allowedCapabilities"`
+}
+
+func (addr *RestrictedAddress) Clone() Address {
+	return &RestrictedAddress{
+		Address:             addr.Address.Clone(),
+		AllowedCapabilities: addr.AllowedCapabilities.Clone(),
+	}
+}
+
+func (addr *RestrictedAddress) VBytes(rentStruct *RentStructure, _ VBytesFunc) VBytes {
+	return rentStruct.VBFactorData.Multiply(VBytes(addr.Size()))
+}
+
+func (addr *RestrictedAddress) ID() []byte {
+	addressID := addr.Address.ID()
+	capabilities := lo.PanicOnErr(CommonSerixAPI().Encode(context.TODO(), addr.AllowedCapabilities))
+
+	// prefix the ID of the underlying address with the AddressType, and append the capabilties
+	return append(append([]byte{byte(AddressRestricted)}, addressID...), capabilities...)
```

Comment:
you are encoding by hand here, why not just encode it with serix?

Reply:
Because the ID() of a MultiAddress is not the serialized form, but the hash of it.

Reply:
why not use `byteutils.ConcatBytes`

Reply:
Oh, didn't know that one. Fixed